zebra-utils/src/bin/scanning-results-reader/main.rs

```diff
@@ -0,0 +1,128 @@
+//! Displays Zebra's scanning results:
+//!
+//! 1. Opens Zebra's scanning storage and reads the results containing scanning keys and TXIDs.
+//! 2. Fetches the transactions by their TXIDs from Zebra using the `getrawtransaction` RPC.
+//! 3. Decrypts the tx outputs using the corresponding scanning key.
+//! 4. Prints the memos in the outputs.
+
+use std::collections::HashMap;
+
+use jsonrpc::simple_http::SimpleHttpTransport;
+use jsonrpc::Client;
+
+use serde_json::value::RawValue;
+use zcash_client_backend::decrypt_transaction;
+use zcash_client_backend::keys::UnifiedFullViewingKey;
+use zcash_primitives::consensus::{BlockHeight, BranchId};
+use zcash_primitives::transaction::Transaction;
+use zcash_primitives::zip32::AccountId;
+use zebra_scan::scan::sapling_key_to_scan_block_keys;
+use zebra_scan::{storage::Storage, Config};
+
+/// Prints the memos of transactions from Zebra's scanning results storage.
+///
+/// Reads the results storage, iterates through all decrypted memos, and prints the them to standard
+/// output. Filters out some frequent and uninteresting memos typically associated with ZECPages.
+///
+/// Notes:
+///
+/// - `#[allow(clippy::print_stdout)]` is set to allow usage of `println!` for displaying the memos.
+/// - This function expects Zebra's RPC server to be available.
+///
+/// # Panics
+///
+/// When:
+///
+/// - The Sapling key from the storage is not valid.
+/// - There is no diversifiable full viewing key (dfvk) available.
+/// - The RPC response cannot be decoded from a hex string to bytes.
+/// - The transaction fetched via RPC cannot be deserialized from raw bytes.
+#[allow(clippy::print_stdout)]
+pub fn main() {
+    let network = zcash_primitives::consensus::Network::MainNetwork;
+    let storage = Storage::new(&Config::default(), zebra_network(&network), true);
+    let mut prev_memo = "".to_owned();
+
+    for (key, _) in storage.sapling_keys_last_heights().iter() {
+        let dfvk = sapling_key_to_scan_block_keys(key, zebra_network(&network))
+            .expect("Scanning key from the storage should be valid")
+            .0
+            .into_iter()
+            .next()
```

Comment:
Optional: There's a method for this:
https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.exactly_one

It also needs a `use itertools::Itertools` and maybe a dependency on `itertools`.

```suggestion
            .exactly_one()
```

Reply:
Do you think we could simplify the return type of `sapling_key_to_scan_block_keys`: https://github.com/zcashfoundation/zebra/blob/a3af2a702d2eb043e89e9b0b8e31e963cb10aeb8/zebra-scan/src/scan.rs#L291-L296 instead?

Reply:
I used it in [9e16ad5](https://github.com/ZcashFoundation/zebra/pull/8104/commits/9e16ad569d13d5a8299d1d0a6a4c791f29b513bb).

Reply:
> Do you think we could simplify the return type of `sapling_key_to_scan_block_keys`: [zcashfoundation/zebra@`a3af2a7`/zebra-scan/src/scan.rs#L291-L296](https://github.com/zcashfoundation/zebra/blob/a3af2a702d2eb043e89e9b0b8e31e963cb10aeb8/zebra-scan/src/scan.rs#L291-L296) instead?

Yes, I think we could remove IVKs from the MVP production code. We don't know the standard IVK format and we don't have code to parse them, so it seems better to simplify it.

Is there a way to move the IVK code into a test? Then we can be sure we have support without running it in production.